src/NHSD.BuyingCatalogue.Identity.Api/Models/ApplicationUser.cs

```diff
@@ -5,14 +5,107 @@ namespace NHSD.BuyingCatalogue.Identity.Api.Models
+        public Guid PrimaryOrganisationId { get; }
+        
+        public OrganisationFunction OrganisationFunction { get; }
+
+        public bool Disabled { get; private set; }
+
+        public bool CatalogueAgreementSigned { get; private set; }
+        
+        private ApplicationUser()
+        {
+        }
+
+        private ApplicationUser(
+            string userName,
+            string firstName, 
+            string lastName, 
+            string phoneNumber, 
+            string email, 
+            OrganisationFunction organisationFunction,
+            Guid primaryOrganisationId) : this()
+        {
+            if (userName is null)
+                throw new ArgumentNullException(nameof(userName));
+
+            if (firstName is null)
+                throw new ArgumentNullException(nameof(firstName));
+
+            if (lastName is null)
+                throw new ArgumentNullException(nameof(lastName));
+
+            if (email is null)
+                throw new ArgumentNullException(nameof(email));
+
+            UserName = userName.Trim();
+            NormalizedUserName = UserName.ToUpperInvariant();
+            FirstName = firstName.Trim();
+            LastName = lastName.Trim();
+            PhoneNumber = phoneNumber?.Trim() ?? throw new ArgumentNullException(nameof(phoneNumber));
+            Email = email.Trim();
+            NormalizedEmail = Email.ToUpperInvariant();
+            OrganisationFunction = organisationFunction ?? throw new ArgumentNullException(nameof(organisationFunction));
+            PrimaryOrganisationId = primaryOrganisationId;
+            Disabled = false;
+            CatalogueAgreementSigned = false;
+        }
+
+        public static ApplicationUser CreateBuyer(
+            string userName,
+            string firstName,
+            string lastName,
+            string phoneNumber,
+            string email,
+            Guid primaryOrganisationId)
+        {
+            return new ApplicationUser(
+                userName, 
+                firstName, 
+                lastName, 
+                phoneNumber, 
+                email, 
+                OrganisationFunction.Buyer, 
+                primaryOrganisationId);
+        }
+
+        public static ApplicationUser CreateAuthority(
+            string userName,
+            string firstName,
+            string lastName,
+            string phoneNumber,
+            string email,
+            Guid primaryOrganisationId)
+        {
+            return new ApplicationUser(
+                userName, 
+                firstName, 
+                lastName, 
+                phoneNumber, 
+                email, 
+                OrganisationFunction.Authority, 
+                primaryOrganisationId);
+        }
+
+        public void MarkAsDisabled()
+        {
+            Disabled = true;
+        }
+
+        public void MarkAsEnabled()
+        {
+            Disabled = false;
+        }
```

Comment:
Slight concern with those names suggest it will actually apply the action i.e. Disable the user including the save to the database, but if you feel this isn't the case happy to change them.

Reply:
Yeah, I see your point. Feel free to leave as is then.